activestorage/app/controllers/active_storage/disk_controller.rb

```diff
@@ -52,6 +52,6 @@ def decode_verified_token
     end
 
     def acceptable_content?(token)
-      token[:content_type] == request.content_mime_type && token[:content_length] == request.content_length
+      Mime::Type.lookup(token[:content_type]) == request.content_mime_type && token[:content_length] == request.content_length
```

Comment:
I think we need to check if `token[:content_type]` is present before calling `Mime::Type.lookup`, otherwise it will raise an exception.

Reply:
Thx for the review!

I'll check if I can write a failing test and see if I can fix accordingly.

Reply:
`.lookup` will now consistently raise `InvalidMimeType` error for `nil`, `""` and invalid strings. I rescue the exception accordingly in the controller. I think that's more correct than checking for presence of the parameter. Wdyt?

Reply:
Disclaimer: I'm not part of any of the [Rails teams](https://rubyonrails.org/community), so I don't really have a say.

I think that this PR now changes 2 different things, to I think that if you split the PR in two it will have greater chances to get accepted.

That said, I'm not sure if https://github.com/rails/rails/pull/50610/files#diff-163405fbf39adb2114465576123b6eceb3512721d5b99e39c589af09110a1ec5R139-R146 should return `:unprocesable_entity`, what is the current behavior?

Reply:
All good, you're raising valid points!

I'll have to think about it again, before my change it would have been possible if both `token[:content_type]` and `request.content_mime_type` could have been nil and pass the validation. I think I changed behavior in this case.